zerver/lib/events.py

```diff
@@ -715,7 +715,7 @@ def apply_event(
 ) -> None:
     if event["type"] == "message":
         state["max_message_id"] = max(state["max_message_id"], event["message"]["id"])
-        if "raw_unread_msgs" in state:
+        if "raw_unread_msgs" in state and "read" not in event["flags"]:
```

Comment:
This is a good bug fix for the production code! (Well, very unlikely case, but nice to have fixed).

I would have put this in its own commit so it gets more emphasis; I'll try extracting it.

Reply:
Ended up deciding it's not worth the effort to extract, so just mentioned it in the commit message.